patches/server/1047-add-disguise-api.patch

```diff
@@ -0,0 +1,274 @@
++        public @org.jetbrains.annotations.NotNull Builder withRightPants(boolean rightPants) {
++            this.rightPants = rightPants;
++            return this;
++        }
++
++        @Override
++        public @org.jetbrains.annotations.NotNull Builder withHat(boolean hat) {
++            this.hats = hat;
++            return this;
++        }
++
++        public @org.jetbrains.annotations.NotNull SkinParts build() {
++            int raw = 0;
++            if (cape) raw |= CAPE;
++            if (jacket) raw |= JACKET;
++            if (leftSleeve) raw |= LEFT_SLEEVE;
++            if (rightSleeve) raw |= RIGHT_SLEEVE;
++            if (leftPants) raw |= LEFT_PANTS;
++            if (rightPants) raw |= RIGHT_PANTS;
++            if (hats) raw |= HAT;
++            return new PaperSkinParts(raw);
++        }
++    }
+ }
+diff --git a/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java b/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1acbaa0a6c681890d4369aafacf974940f925a9b
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java
+@@ -0,0 +1,84 @@
++package io.papermc.paper.disguise;
++
++import com.destroystokyo.paper.profile.PlayerProfile;
++import java.util.EnumSet;
++import java.util.List;
++import net.minecraft.network.protocol.game.ClientboundAddEntityPacket;
++import net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket;
++import net.minecraft.network.syncher.EntityDataAccessor;
++import net.minecraft.network.syncher.SynchedEntityData;
++import net.minecraft.server.level.ServerPlayer;
++import net.minecraft.world.entity.Entity;
++import net.minecraft.world.entity.EntityType;
++import net.minecraft.world.entity.player.Player;
++import net.minecraft.world.phys.Vec3;
++import org.bukkit.Bukkit;
++
++public final class DisguiseUtil {
++    private DisguiseUtil(){}
++
++    public static boolean tryDisguise(ServerPlayer player, Entity entity, ClientboundAddEntityPacket clientboundAddEntityPacket) {
++            return switch (entity.getBukkitEntity().getDisguiseData()) {
++                case DisguiseData.OriginalDisguise disguise -> false;
++                case io.papermc.paper.disguise.EntityTypeDisguise(var type) -> {
++                    player.connection.send(create(clientboundAddEntityPacket, org.bukkit.craftbukkit.entity.CraftEntityType.bukkitToMinecraft(type)));
++                    yield  true;
++                }
++                case io.papermc.paper.disguise.PlayerDisguise(var playerProfile, var listed, var skinParts) -> {
++                    PlayerProfile adapted = Bukkit.createProfile(entity.getUUID(), entity.hasCustomName()? entity.getBukkitEntity().getCustomName(): playerProfile.getName());
++                    adapted.setProperties(playerProfile.getProperties());
++                    net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket.Entry playerUpdate =
```

Comment:
some of the classes here are imported and some use the fully qualified name (even if they are imported)
just use imports for everything, its a class we add so no risk of dum conflicts